crates/oxc_cli/src/lib.rs

```diff
@@ -61,6 +61,17 @@ impl Cli {
             })
             .sum();
 
+        // https://eslint.org/docs/latest/use/command-line-interface#--max-warnings
+        if self.cli_options.max_warnings > -1 {
+            let max_warnings = self.cli_options.max_warnings as usize;
+            if number_of_diagnostics > max_warnings {
```

Comment:
Technically `number_of_diagnostics` could encapsulate non-warning diagnostics in the future, in which case we'd probably want to introduce a separate variable tracking just warnings for this CLI option

Reply:
Yep this is fine for now.